tasks/fs-default.yml

```diff
@@ -42,14 +42,27 @@
          and ['util-linux'] is not subset(ansible_facts.packages.keys()) and
          util_linux_installed is undefined and not ansible_check_mode"
 
-- name: unmount fs if we're going to reformat
+- name: Check if we are trying to format the same fs-type as the current device fs-type
+  set_fact:
+    same_fs_flag: True
+  when: volume.fs_type == volume._orig_fs_type
```

Comment:
In `defaults/main.yml`, we specify `volume._orig_fs_type: ""`, and then update that scoped variable as needed in `tasks/volume-default.yml` when we run a blkid command to find a device's current fs-type, and a successful rc is returned. We set a flag to indicate we are passing the same fs-type as the one on an existing disk device we are trying to configure.